vllm/sequence.py

```diff
@@ -489,6 +490,19 @@ def __repr__(self) -> str:
                 f"num_blocks={self.n_blocks}, ")
 
 
+@dataclass
+class SequenceGroupState:
+    """Mutable state tied to a specific sequence group"""
```

Comment:
I wonder if this should be a part of SequenceData? All the states are now stored in SequenceData now actually

Reply:
also IIUC `num_steps ` is not state, and it may not belong here?

Reply:
perhaps eventually we will support different num_steps in a batch and so we track this per SequenceGroup. There used to be a SequenceGroupState used for seed generator but has since been removed. https://github.com/vllm-project/vllm/pull/6698